src/lib/flow/sol-flow-builder.c

```diff
@@ -973,8 +973,9 @@ sol_flow_builder_get_node_type(struct sol_flow_builder *builder)
     err = add_guards(builder);
     if (err < 0) {
         SOL_WRN("Failed to allocate memory for constructing node type");
+        free(opts);
         errno = ENOMEM;
-        return NULL;
+        goto guards_error;
```

Comment:
`guards_error` label already has a `free(opts)`, so isn't this causing a double free, since there's a `free(opts)` just after the `SOL_WRN...`?